### ISSUE TITLE
RBAC impersonation isn't working as expected with OIDC integration

### ISSUE THREAD

**Issue by vkukul-zendesk** [karmada-io/karmada#3176]
**What happened**:
RBAC impersonation isn't working as expected with Okta integration. RBAC policies exist in member cluster and member cluster is in pull mode. Cluster-proxy clusterrole and clusterrolebinding have been updated to include required groups. Requests route through karmada-search component, but don't return expected resources.

**What you expected to happen**:
Have requests from karmada-apiserver respect RBAC policies in the member cluster.

**How to reproduce it (as minimally and precisely as possible)**:
Bind policies to an Okta group on the member cluster and try to validate those policies through the karmada api server.

**Anything else we need to know?**:
We have one member cluster with the ANP proxy in pull mode. Karmada api-server has the following flags for okta.
```
- --oidc-issuer-url=issuer-url
- --oidc-client-id=client-id
- --oidc-username-claim=email
- '--oidc-username-prefix=okta:'
- --oidc-groups-claim=groups
- '--oidc-groups-prefix=okta:'
```

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
```
karmadactl version: version.Info{GitVersion:"v1.4.0-196-gfed2f269-dirty", GitCommit:"fed2f269f5a47e4d37db8bffb29aabe4641112d4", GitTreeState:"dirty", BuildDate:"2023-01-31T16:44:56Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/arm64"}
```
- Others:

**Comment by RainbowMango** [comment]
@XiShanYongYe-Chang Could you please take a look?

**Comment by XiShanYongYe-Chang** [comment]
> Requests route through karmada-search component, but don't return expected resources.

Hi @vkukul-zendesk, using RBAC impersonation is supposed to be done through `karmada aggregated apiserver` component, I'm not sure if it's wrong with your description.

And can I ask you some questions: what is the URL for accessing the resource? What user did you use to access, and can you provide specific information about the error returned by the access?

**Comment by a7i** [comment]
Hi @XiShanYongYe-Chang 

Given that:
- my member cluster is `member1`
- my okta group is `support`
- the `support` okta group has admin user impersonation + access to delete pods in namespace `flink-support-pipeline`
- `cluster-proxy-clusterrole` definition RBAC:
<details><code>
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: cluster-proxy-clusterrole
    rules:
    - apiGroups:
      - 'cluster.karmada.io'
      resources:
      - clusters/proxy
      resourceNames:
      - member1
      verbs:
      - '*'
    - apiGroups:
      - search.karmada.io
      resourceNames:
      - karmada
      resources:
      - proxying/proxy
      verbs:
      - '*'
    ---
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: cluster-proxy-clusterrolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: cluster-proxy-clusterrole
    subjects:
      - apiGroup: rbac.authorization.k8s.io
        kind: User
        name: system:admin
      - apiGroup: rbac.authorization.k8s.io
        kind: Group
        name: support
</code></details>


## Direct member1 cluster access
```bash
kubectl auth can-i delete pods --namespace flink-support-pipeline --as admin --as-group support --as-group system:authenticated --context member1

yes
```

```bash
kubectl auth can-i delete pods --namespace kube-system --as admin --as-group support --as-group system:authenticated --context member1

no
```

## Karmada API access via [Proxy](https://karmada.io/docs/userguide/globalview/proxy-global-resource/#step2-access-by-proxy)
```
kubectl auth can-i delete pods --namespace flink-support-pipeline --as admin --as-group support --as-group system:authenticated --context karmada-apiserver

yes
```

```
kubectl auth can-i delete pods --namespace kube-system --as admin --as-group support --as-group system:authenticated --context karmada-apiserver

yes
```
   ^^^ **Incorect**

**Comment by vkukul-zendesk** [comment]
### Karmada API access via member cluster
Given that we us the URL `https://${EXT_API_URL}/apis/cluster.karmada.io/v1alpha1/clusters/member1/proxy`

```
kubectl auth can-i delete pods --namespace flink-support-pipeline --as admin --as-group support --as-group system:authenticated --context member1

yes
```

```
kubectl auth can-i delete pods --namespace kube-system --as admin --as-group support --as-group system:authenticated --context member1

no
```

Are we able to achieve the same RBAC consistency utilizing the [Proxy](https://karmada.io/docs/userguide/globalview/proxy-global-resource/#step2-access-by-proxy) without targeting a specific member cluster. We want to provide a level of abstraction on the cluster name and number of member clusters to abstract away from the user kubeconfig so they can use a single kubeconfig.

**Comment by XiShanYongYe-Chang** [comment]
Hi @vkukul-zendesk @a7i, I'm sorry to get back to you today. I was busy with other things yesterday. Let me understand what you mean and correct me if I'm wrong.

You want to use `search.karmada.io proxying/proxy` the same way as `cluster.karmada.io cluster/proxy`, unified authentication is provided during access to member clusters.

Currently, karmada does not support this capability, and I agree with you that karmada needs to have this capability.

**Comment by a7i** [comment]
Thanks @XiShanYongYe-Chang thank you. Your confirmation proves our theory.

Do you have a timeline for when this will be supported? If you can provide us with some hints we could try to take a stab at it.

**Comment by RainbowMango** [comment]
@a7i @vkukul-zendesk Thanks for spotting this, I suppose we can put it into v1.6's backlog. (which should be released: by the end of May).

Please let me know if it fits your plan, and I'm looking forward to hearing your use case that significantly affects its priority.

**Comment by vkukul-zendesk** [comment]
Hi @RainbowMango sorry for the late reply, but below is what we are trying to accomplish.

Objective 1:
We would like to use `search.karmada.io` proxying/proxy to achieve unified authentication the same way as `cluster.karmada.io` cluster/proxy with user impersonation. Unlike `cluster.karmada.io` though, we don’t want kubeconfigs to be member cluster specific with the server address as we intend to have ephemeral clusters that may last a few weeks.

Objective 2:
The team’s goal with Karmada is to utilize `search.karmada.io` to provide our already existing Okta groups and users with access to member clusters through the [Proxy](https://karmada.io/docs/userguide/globalview/proxy-global-resource/#step2-access-by-proxy) with user impersonation without targeting a specific member cluster.

**Comment by XiShanYongYe-Chang** [comment]
Hi @vkukul-zendesk @a7i , I'd like to reconfirm:

For the [Proxy](https://karmada.io/docs/userguide/globalview/proxy-global-resource/#step2-access-by-proxy), the `get`, `list`, and `watch` requests depend on the cache on the Karmada control plane. These requests may not support unified authentication. That is, as long as a user has the `search/proxy` permission, the user can query all resources.

Unified authentication is only feasible for requests that can be proxied to member clusters, such as `update`, `patch`, `delete`, etc.

**Comment by XiShanYongYe-Chang** [comment]
Hi @vkukul-zendesk @a7i , I've implemented search proxy unified auth with the PR #3279, would you like to have a test with it?

**Comment by XiShanYongYe-Chang** [comment]
> For the [Proxy](https://karmada.io/docs/userguide/globalview/proxy-global-resource/#step2-access-by-proxy), the get, list, and watch requests depend on the cache on the Karmada control plane. These requests may not support unified authentication. That is, as long as a user has the search/proxying permission, the user can query all resources.
>
> Unified authentication is only feasible for requests that can be proxied to member clusters, such as update, patch, delete, etc.

Another thing we might note is that permission verification using the `kubectl auth can-i` command does not work now, the correct authentication is performed only when the operation is performed on a specific resource.
